lib/Controller/ApiController.php

```diff
@@ -88,12 +110,45 @@ public function __construct($appName,
 		$this->sessionStatistics  = $sessionStatistics;
 	}
 
+	private function checkAuthorized() {
+		$token = $this->request->getHeader('NC-Token');
+		if (!empty($token)) {
+			$storedToken = $this->config->getAppValue('serverinfo', 'token', null);
+			if ($storedToken === $token) {
```

Comment:
This should use https://www.php.net/manual/en/function.hash-equals.php

Reply:
thanks, done